### PR TITLE
[ML] Fixed Various Schema Issues

### PR DESCRIPTION
**Pull request by nthandeMS**
# Description
- Changed `services` field for jobs to use an enum.
- Added datastore arm_type to datastore fields in job inputs and environments.
- Updated regex for azureml schemas from `^azureml:\w+` to `^azureml:.+` to include greater variety of ARM types.
- Added stricter check to ArmStr when parsing `azureml:` prefix.
- Added local path arm_type to LocalPathField for VS Code autocompletion.
  - Fixes issue with `code` field in Command Job not supporting local path completion.
- Explicitly added ISO datetime format to utc_time_created field in models.
- Formatting fixes applied by `black`.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/assets/model.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ class ModelSchema(PathAwareSchema):
     description = fields.Str()
     properties = fields.Dict()
     tags = fields.Dict()
-    utc_time_created = fields.DateTime()
+    utc_time_created = fields.DateTime(format="iso")
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/assets/model.py:37]: This should be a `dump_only` field right?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/core/fields.py [hunk 2]
```diff
@@ -247,13 +271,15 @@ def _validate(self, value):
...
+            raise ValidationError(
+                f"Not a valid ISO8601-formatted datetime string: {value}"
+            )
 
 
 class ArmStr(Field):
     def __init__(self, **kwargs):
         self.azureml_type = kwargs.pop("azureml_type", None)
-        self.pattern = kwargs.pop("pattern", "^azureml:.*")
+        self.pattern = kwargs.pop("pattern", r"^azureml:[\w\/-]+")
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/core/fields.py:282]: "."  is a legal character in ARM ids. Does \w\ allow it?

Comment (nthandeMS) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/core/fields.py:282]: It does not. After discussing with the VS Code team, we decided to replace this `[\w\/-]` with `.` similar to what we originally had -- the different paths will be parsed by the VS Code extension.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/assets/model.py (hunk 1 above)
Comment (nthandeMS) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/assets/model.py]: Yes, good catch.

**Review by harneetvirk — APPROVED** [review]

**Pull request merged**